adagucserverEC/CSLD.cpp

```diff
@@ -0,0 +1,305 @@
+#include "CDebugger.h"
+#include "CSLD.h"
+#include "../hclasses/CXMLParser.h"
+#include "../hclasses/CHTTPTools.h"
+
+const char *CSLD::className = "CSLD";
+
+CSLD::CSLD() {
+  //SETUP STYLE variables
+  this->myOwnStyle = new CServerConfig::XMLE_Style();
+  this->myOwnLegend = new CServerConfig::XMLE_Legend();
+}
+
+void CSLD::setServerParams(CServerParams *serverParams) {
+  this->serverParams = serverParams;
+  this->serverConfig = serverParams->cfg;
+}
+
+//Start processing the SLD url
+int CSLD::processSLDUrl(CT::string sldUrl) {
+  CServerParams cServerParams;
+
+  //Check if url extension is .xml
+  if(!sldUrl.endsWith(".xml")){
```

Comment:
Check for startsWith("http"), ensures that no local file path is specified instead